app/controllers/sessions_controller.rb

```diff
@@ -10,25 +10,45 @@ def can_authenticate?(user)
 
   def create
     @user = User.find_by(email: params.dig(:session, :email)&.downcase)
+
     if can_authenticate?(@user)
-      forwarding_url = session[:forwarding_url]
-      reset_session
-      log_in @user
-      params[:session][:remember_me] == '1' ? remember(@user) : forget(@user)
-      session[:session_token] = @user.session_token
-      redirect_to forwarding_url || @user
-    else
-      flash.now[:danger] = 'Invalid email/password combination'
-      render 'new'
+      if @user.activated?
```

Comment:
can you extract the logic in the if & else branches into methods to improve readability since we have nested conditionals here?